apps/api_web/lib/api_web/params.ex

```diff
@@ -256,23 +256,60 @@ defmodule ApiWeb.Params do
       ...>      }
       ...>    }
       iex(2)> ApiWeb.Params.filter_params(params, ["sort", "route"])
-      %{"sort" => "4,5,6", "route" => "1,2,3"}
+      {:ok, %{"sort" => "4,5,6", "route" => "1,2,3"}}
 
       iex> ApiWeb.Params.filter_params(%{"sort" => "1,2,3"}, [])
-      %{}
+      {:ok, %{}}
 
   """
-  @spec filter_params(map, [String.t()]) :: map
+  @spec filter_params(map, [String.t()]) :: {:ok, map} | {:error, atom}
   def filter_params(params, keys) do
-    params
-    |> Map.take(keys)
-    |> Map.merge(json_api_filter_params(params, keys))
+    with {:ok, filtered} <- validate_filters(params, keys) do
+      result =
+        params
+        |> Map.take(keys)
+        |> Map.merge(filtered)
+
+      {:ok, result}
+    else
+      {:error, _} = error -> error
+    end
   end
 
-  defp json_api_filter_params(params, keys) do
+  @spec validate_filters(map, [String.t()]) :: {:ok, map} | {:error, atom}
+  def validate_filters(params, keys) do
     case params["filter"] do
-      filter when is_map(filter) -> Map.take(filter, keys)
-      _ -> %{}
+      filter when is_map(filter) ->
+        filtered = Map.take(filter, keys)
+
+        if map_size(filtered) == map_size(filter) do
+          {:ok, filtered}
+        else
+          {:error, :bad_filter}
```

Comment:
Can you include the bad filter values? It'll be a better error message.

Reply:
Done